python/lsst/validate/drp/matchreduce.py

```diff
@@ -212,29 +256,21 @@ def _loadAndMatchCatalogs(repo, dataIds, matchRadius,
 
     for vId in dataIds:
 
-        if useJointCal:
-            try:
-                photoCalib = butler.get("jointcal_photoCalib", vId)
-            except (FitsError, dafPersist.NoResults) as e:
-                print(e)
-                print("Could not open photometric calibration for ", vId)
-                print("Skipping this dataId.")
-                continue
+        if doApplyExternalPhotoCalib:
```

Comment:
At this point, I think we should pull these `if` statements into separate methods `loadPhotoCalib`, `loadSkyWcs`. That will help clean this overly-long method up.